### PR TITLE
Create file for mutant creator test during test run

### PR DESCRIPTION
**Pull request by BackEndTea**
This PR:
Fixes: #248

### CODE REVIEW

File: tests/Mutant/MutantCreatorTest.php [hunk 1]
```diff
@@ -18,6 +18,28 @@
 
 class MutantCreatorTest extends MockeryTestCase
 {
+    private $fileLocation;
+
+    public function setUp()
+    {
+        parent::setUp();
+        $this->fileLocation = __DIR__ . '/../Fixtures/MutantCreator/HasMutation/mutant.hash.infection.php';
```
Comment (sanmai) [inline tests/Mutant/MutantCreatorTest.php]: If it does create a file, this better be a temporary directory. [Should be trivial.](https://github.com/infection/infection/blob/79c873663e088a3cbaf3a7a1cd7600f25845fb90/tests/TestFramework/PhpSpec/Config/Builder/InitialConfigBuilderTest.php#L36)

File: tests/Mutant/MutantCreatorTest.php [hunk 2]
```diff
@@ -18,10 +18,39 @@
 
 class MutantCreatorTest extends MockeryTestCase
 {
-    public function test_it_uses_avaialable_file_if_hash_is_the_same()
+    /**
+     * @var string
+     */
+    private $fileName = '/mutant.hash.infection.php';
```
Comment (borNfreee) [inline tests/Mutant/MutantCreatorTest.php]: If it's not being (re)set anywhere, it's a `const`, not variable

File: tests/Mutant/MutantCreatorTest.php [hunk 3]
```diff
@@ -42,10 +71,10 @@ public function test_it_uses_avaialable_file_if_hash_is_the_same()
         $coverage->shouldReceive('hasExecutedMethodOnLine')->andReturn(true);
         $coverage->shouldReceive('getAllTestsFor')->andReturn(['test', 'list']);
 
-        $creator = new MutantCreator($tempdir, $differ, $standard);
+        $creator = new MutantCreator($this->directory, $differ, $standard);
         $mutant = $creator->create($mutation, $coverage);
 
-        $this->assertSame($tempdir . '/mutant.hash.infection.php', $mutant->getMutatedFilePath());
+        $this->assertSame($this->directory . '/mutant.hash.infection.php', $mutant->getMutatedFilePath());
```
Comment (borNfreee) [inline tests/Mutant/MutantCreatorTest.php]: please replace `/mutant.hash.infection.php` with $this->fileName (or self::FILE_NAME)

**Comment by borNfreee** [comment]
Thank you @BackEndTea

**Comment by BackEndTea** [comment]
Thank you @borNfreee

**Pull request merged**